src/App/Fossa/Analyze/VulnerabilityReport.hs

```diff
@@ -0,0 +1,138 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.Analyze.VulnerabilityReport (
+  logVulnerabilityReport,
+) where
+
+import App.Fossa.Analyze.Project (ProjectResult (..))
+import Control.Algebra (Has)
+import Control.Effect.Diagnostics qualified as Diag (Diagnostics)
+import Data.Foldable (traverse_)
+import Data.List qualified as List
+import Data.Maybe qualified
+import Data.Set qualified as Set
+import Data.String.Conversion (toString, toText)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import DepTypes (Dependency (..), VerConstraint)
+import Effect.Logger (
+  Logger,
+  logInfo,
+ )
+import Graphing (directList, getRootsOf, vertexList)
+import Path (File, SomeBase)
+import Prettyprinter (Pretty (pretty), vsep)
+import Srclib.Converter (verConstraintToRevision)
+import Text.PrettyPrint.Boxes qualified as Boxes
+
+data VulnerableDependency = VulnerableDependency
+  { vdName :: Text
+  , isVulnerable :: Maybe VerConstraint -> Bool
+  }
+
+data VulnerableDependencyOrigin
+  = VulnerableDependencyManifestFiles [SomeBase File]
+  | VulnerableDependencyRootDependency [Dependency]
+  deriving (Eq, Ord)
+
+instance Show VulnerableDependencyOrigin where
+  show (VulnerableDependencyManifestFiles files) = toString $ Text.intercalate ", " (map (toText . show) files)
+  show (VulnerableDependencyRootDependency deps) = toString (Text.intercalate ", " $ map dependencyName deps)
+
+data VulnerableReportItem = VulnerableReportItem
+  { vriDep :: Dependency
+  , vriDepOriginDep :: VulnerableDependencyOrigin
+  }
+  deriving (Eq, Ord)
+
+-- HACK - Please FIX ASAP!
+-- Since we do not have built in version evaluator system, nor we parse version text into representative form.
+-- This is simplest of hack, ideally each versions are listed as opposed to providing evaluator functions.
+allVersionsBelow2_17 :: Maybe VerConstraint -> Bool
```

Comment:
- [ ] 😱 Add unit test for different maven repo 😱
 
 Techdebt: Add evaluation engine for VerConstraint?